core/src/conn.rs

```diff
@@ -43,6 +43,7 @@ impl MongoConnection {
         login_timeout: Option<u32>,
     ) -> Result<Self> {
         client_options.connect_timeout = login_timeout.map(|to| Duration::new(to as u64, 0));
+        client_options.max_pool_size = Some(5);
```

Comment:
I worked on this in the past, and 100 is almost always the wrong answer for most people. As a rough rule, it should be around 2 threads per core + 1 for concurrent use. I didn't think we were doing that many concurrent ops though, since it's fairly linear in what it calls.

As discussed in Slack, I'll revert this change and we can discuss it in more detail later!